frontend/src/Reducers/productReducer.js

```diff
@@ -0,0 +1,69 @@
+const initialProductData = {
+  priceRange: { min: 0, max: 9999 },
+  sortByLowAndHigh: "",
+  categoriesList: {},
+  sortByRating: "",
+  productsList: [],
+  loading: true,
+};
+
+const productReducer = (state, action) => {
+  switch (action.type) {
+    case "INITIALIZE_DATA":
+      const categories = action.payload.categories.reduce(
+        (acc, cat) => (acc[cat] ? acc : { ...acc, [cat]: false }),
+        {}
+      );
+      return {
+        ...state,
+        productsList: action.payload.products,
+        categoriesList: categories,
+        loading: false,
+      };
+
+    case "PRICE_RANGE":
+      return {
+        ...state,
+        priceRange: action.payload,
+      };
+
+    case "SORT_BY_HIGH_LOW":
+      return {
+        ...state,
+        sortByHighLow: action.payload,
+      };
+
+    case "CATEGORY":
+      return {
+        ...state,
+        categoriesList: {
```

Comment:
See, here if you can improve the naming of `categoriesList`. As you have used `List` in the name, so I thought it must be an array, but you are using an object. 

So, see if it can be improved.